app/views/posts/index.html.erb

```diff
@@ -1,3 +1,18 @@
-<h1>Latest Posts</h1>
-<%= render 'list' %>
+<% if params[:search].present? %>
+  <% if @posts.size == 0 %>
+    <div class="alert alert-info">
+      <button type="button" class="close" data-dismiss="alert">&times;</button>
+      <span>No result found</span>
+    </div>
+    <h1>Latest Posts</h1>
+    <% @posts = Post.all %>
```

Comment:
Yes, you should not call `Post.all` in the view. This belongs in the controller. If there are no results from the search, it should not display any posts. It would also be nice to display the search terms on the page, even when there are no results. Like this:

```erb
<% if params[:search].present? %>
  <h1>Post Search: "<%= params[:search] %>"</h1>
  <% if @posts.size == 0 %>
    <div class="alert alert-info">
      <button type="button" class="close" data-dismiss="alert">&times;</button>
      <span>No result found</span>
    </div>
  <% end %>
<% else %>
  <h1>Latest Posts</h1>
<% end %>
```